hazelcast/src/main/java/com/hazelcast/internal/management/dto/PermissionConfigDTO.java

```diff
@@ -0,0 +1,180 @@
+
+    private PermissionConfig permissionConfig;
+
+    public PermissionConfigDTO() {
+
+    }
+
+    public PermissionConfigDTO(PermissionConfig permissionConfig) {
+        this.permissionConfig = permissionConfig;
+    }
+
+    @Override
+    public JsonObject toJson() {
+        JsonObject object = new JsonObject();
+        object.add("permissionType", permissionConfig.getType().getNodeName());
+        object.add("name", permissionConfig.getName());
+        if (StringUtil.isNullOrEmptyAfterTrim(permissionConfig.getPrincipal())) {
+            object.add("principal", "*");
+        } else {
+            object.add("principal", permissionConfig.getPrincipal());
+        }
+
+        Set<String> endpoints = permissionConfig.getEndpoints();
+        if (endpoints != null) {
+            JsonArray endpointsArray = new JsonArray();
+            for (String endpoint : endpoints) {
+                endpointsArray.add(endpoint);
+            }
+            object.add("endpoints", endpointsArray);
+        }
+
+        Set<String> actions = permissionConfig.getActions();
+        if (actions != null) {
+            JsonArray actionsArray = new JsonArray();
+            for (String action : actions) {
+                actionsArray.add(action);
+            }
+            object.add("actions", actionsArray);
+        }
+
+        return object;
+    }
+
+    @Override
+    public void fromJson(JsonObject json) {
+        permissionConfig = new PermissionConfig();
+        permissionConfig.setType(PermissionConfig.PermissionType.getType(json.getString("permissionType", null)));
+        permissionConfig.setName(json.get("name").asString());
+        permissionConfig.setPrincipal(json.getString("principal", "*"));
+
+        JsonValue endpointsVal = json.get("endpoints");
+        if (endpointsVal != null) {
+            Set<String> endpoints = new HashSet<String>();
+            for (JsonValue endpoint : endpointsVal.asArray().values()) {
+                endpoints.add(endpoint.asString());
+            }
+            permissionConfig.setEndpoints(endpoints);
+        }
+
+        JsonValue actionsVal = json.get("actions");
+        if (actionsVal != null) {
+            Set<String> actions = new HashSet<String>();
+            for (JsonValue action : actionsVal.asArray().values()) {
+                actions.add(action.asString());
+            }
+            permissionConfig.setActions(actions);
+        }
+
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(permissionConfig.getType().getNodeName());
+        out.writeUTF(permissionConfig.getName());
+        if (StringUtil.isNullOrEmptyAfterTrim(permissionConfig.getPrincipal())) {
+            out.writeUTF("*");
+        } else {
+            out.writeUTF(permissionConfig.getPrincipal());
+        }
+
+        Set<String> endpoints = permissionConfig.getEndpoints();
+        if (endpoints != null) {
```

Comment:
`permissionConfig.getEndpoints` works with get-or-create semantics so the `null` check here is redundant. It's the same for `permissionConfig.getActions` below also.